src/components/CategoryHeading.svelte

```diff
@@ -0,0 +1,32 @@
+<script>
+  import { areAllDefined } from "../util/genUtils";
+  import { renderPercentage, renderCatHeadString } from "../helpers/categoryHelper"; 
+  export let variableData
+  export let variable
+  export let category
+  export let location;
+</script>
+
+<h2>
+  {#if areAllDefined([variableData, variable, category, location])}
+    <div>
+      <span>
+        {renderPercentage(variableData[category?.code]?.percentage)}<span>%</span>
+      </span>
+    </div>
+    <div>
+      <div>
+        <span>
+          {renderCatHeadString(variable, category, location, category.category_h_pt2)}
+        </span>
+      </div>
+      <div>
+        <span>
+          {renderCatHeadString(variable, category, location, category.category_h_pt3)}
+        </span>
+      </div>
+    </div>
+  {:else}
+    <span>loading...</span>
```

Comment:
would it be better to render the actual component html even as it's loading?  that's the pattern we've used in the CensusTable component, to avoid a jumpy UI

Reply:
You end up with loads of ternaries everywhere though, and you'll be rendering empty divs if theres nothing to load

Reply:
@petem-methods - your call

Reply:
Defo would remove the "loading" stuff. The UI shouldn't jump and loading indication should be handled out-of-bounds of individual components.

Reply:
done